cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -343,4 +445,91 @@ private ProfileId getResolvedProfileId(EntityId entityId) throws IOException {
     }
     return programIds;
   }
+
+  /**
+   * Get the schedule identified by the given schedule id
+   */
+  private ScheduleDetail getSchedule(ScheduleId scheduleId) throws IOException, NotFoundException {
+    String url = String.format("namespaces/%s/apps/%s/versions/%s/schedules/%s",
+                               scheduleId.getNamespace(), scheduleId.getApplication(), scheduleId.getVersion(), scheduleId.getSchedule());
+    HttpResponse httpResponse = runHttpGet(url);
+    ObjectResponse<ScheduleDetail> objectResponse =
+      ObjectResponse.fromJsonBody(httpResponse, new TypeToken<ScheduleDetail>() { }.getType(), GSON);
+    return objectResponse.getResponseObject();
+  }
+
+  /**
+   * Get the list of schedules for the given program id
+   */
+  private List<ScheduleDetail> listSchedules(ProgramId programId) throws IOException {
+    String url = String.format("namespaces/%s/apps/%s/versions/%s/schedules",
+                               programId.getNamespace(), programId.getApplication(), programId.getVersion());
+    HttpResponse httpResponse = null;
+    try {
+      httpResponse = runHttpGet(url);
+    } catch (Exception e) {
+      throw new IOException(
+        String.format("Failed to list schedules for program %s", programId.toString()), e);
+    }
+    ObjectResponse<List<ScheduleDetail>> objectResponse = ObjectResponse.fromJsonBody(
+      httpResponse, new TypeToken<List<ScheduleDetail>>() { }.getType(), GSON);
+    return objectResponse.getResponseObject();
+  }
+
+  /**
+   * Get the metadata of the schedule identified by the given schedule id
+   */
+  private ScheduleMetadata getScheduleMetadata(ScheduleId scheduleId) throws IOException, NotFoundException {
+    String url = String.format("namespaces/%s/apps/%s/versions/%s/schedules/%s/metadata",
+                               scheduleId.getNamespace(), scheduleId.getApplication(), scheduleId.getVersion(), scheduleId.getSchedule());
+    HttpResponse httpResponse = runHttpGet(url);
+    ScheduleMetadata metadata = GSON.fromJson(httpResponse.getResponseBodyAsString(), ScheduleMetadata.class);
+    return metadata;
+  }
+
+
+  /**
+   * Get the metadata of the application identified by the given id
+   */
+  private ApplicationMeta getApplicationMeta(ApplicationId appId) throws IOException, NotFoundException {
+    String url = String.format("namespaces/%s/apps/%s/versions/%s/metadata",
+                               appId.getNamespace(), appId.getApplication(), appId.getVersion());
+    HttpResponse resp = runHttpGet(url);
+    ApplicationMeta meta = GSON.fromJson(resp.getResponseBodyAsString(), ApplicationMeta.class);
+    return meta;
+  }
+
+  /**
+   * Get the metadata of all applications in the given namespace
+   */
+  private List<ApplicationMeta> getAllApplicationMetadata(String namespace) throws IOException {
```

Comment:
All these methods should better be abstract out into separate classes/interfaces. If defined as interfaces, it also gives us the flexibility on the implementation via guice bindings. (e.g. in sandbox is just a in-memory call, vs in distributed env is a network call).